### PR TITLE
chore(ci_visibility): fixes for v2 pytest plugin

### PR DESCRIPTION
**Pull request by romainkomorndatadog**
This makes a few fixes to tests and the pytest v2 plugin to properly gate around CI Visibility enablement as well as make sure that the service is disabled after tests to prevent state leaks between tests that can cause flakiness based on ordering.

## Checklist
- [x] PR author has checked that all the criteria below are met
- The PR description includes an overview of the change
- The PR description articulates the motivation for the change
- The change includes tests OR the PR description describes a testing strategy
- The PR description notes risks associated with the change, if any
- Newly-added code is easy to change
- The change follows the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html)
- The change includes or references documentation updates if necessary
- Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist
- [x] Reviewer has checked that all the criteria below are met 
- Title is accurate
- All changes are related to the pull request's stated goal
- Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- Testing strategy adequately addresses listed risks
- Newly-added code is easy to change
- Release note makes sense to a user of the library
- If necessary, author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

File: tests/contrib/pytest/test_pytest.py [hunk 1]
```diff
@@ -475,7 +475,7 @@ def test_xfail_conditional():
         """
         )
         file_name = os.path.basename(py_file.strpath)
-        rec = self.inline_run("--ddtrace", file_name)
+        rec = self.inline_run("--ddtrace", "-s", "-vvv", file_name)
```
Comment (romainkomorndatadog) [inline tests/contrib/pytest/test_pytest.py:478]:
```suggestion
        rec = self.inline_run("--ddtrace", file_name)
```

File: tests/ci_visibility/api/test_internal_test_visibility_api.py [hunk 2]
```diff
@@ -13,6 +15,22 @@ class TestCIITRMixin:
     Note: these tests do not bother discovering a session as the ITR functionality currently does not rely on sessions.
     """
 
+    @pytest.fixture(scope="function", autouse=True)
+    def _disable_ci_visibility(self):
+        try:
+            if CIVisibility.enabled:
+                CIVisibility.disable()
```
Comment (vitor-de-araujo) [inline tests/ci_visibility/api/test_internal_test_visibility_api.py:22]: What do you think about making `CIVisibility.disable()` idempotent so we don't have to always check if it's enabled first?

**Review by vitor-de-araujo — APPROVED** [review]

File: tests/ci_visibility/api/test_internal_test_visibility_api.py (hunk 2 above)
Comment (romainkomorndatadog) [reply tests/ci_visibility/api/test_internal_test_visibility_api.py:22]: I think it's a great idea for another PR...

**Pull request merged**